src/WorldMap.cpp

```diff
@@ -40,3 +235,200 @@ void WorldMap::clearMap()
 {
 	m_tileLayers.clear();
 }
+
+void WorldMap::addTileSheet(const std::string& tileSheetPath, const TileSheetDetails & tileSheetDetails)
+{
+	assert(!hasTileLayer(tileSheetDetails.m_name));
+	m_tileSheets.emplace(tileSheetDetails.m_name, TileSheet(tileSheetDetails, tileSheetPath));
+}
+
+void WorldMap::parseTileMap(const TiXmlElement & root, const LevelDetails & levelDetails)
+{
+	const int tileLayerCount = getNumberOfTileLayers(root);
+	const TiXmlElement* tileLayerNode = findNode(root, "layer"); //Get the first layer node
+	assert(tileLayerNode);
+	for (int i = 1; i <= tileLayerCount; ++i)
+	{
+		//If node in question is a layer 
+		if (tileLayerNode->Value() == std::string("layer") &&
+			tileLayerNode->Attribute("name") == std::string("Tile Layer " + std::to_string(i)))
+		{
+			std::string tileLayerName = tileLayerNode->Attribute("name");
+			//assert(tileLayerNode->Attribute("tilesheetname"));
+			std::string tileSheetName = "tilesheet_complete";
+
+			if (tileLayerNode->FirstChildElement()->Value() == std::string("data") ||
+				tileLayerNode->FirstChildElement()->NextSiblingElement() != 0 && tileLayerNode->FirstChildElement()->NextSiblingElement()->Value() == std::string("data"))
+			{
+				assignTileLayer(decodeTileLayer(*tileLayerNode, levelDetails), levelDetails.m_mapSize, tileSheetName, std::move(tileLayerName));
+			}
+		}
+		if (tileLayerNode->NextSiblingElement())
+		{
+			//Go to the next layer
+			tileLayerNode = tileLayerNode->NextSiblingElement();
+		}
+	}
+}
+
+void WorldMap::parseTileSheets(const TiXmlElement & root)
+{
+	const TiXmlElement* tileSetNode = findNode(root, "tileset");
+	assert(tileSetNode);
+	bool continueSearching = true;
+	while (continueSearching)
```

Comment:
This is not a great loop structure - it introduces an unnecessary variable.   Could just do:

while (tileSetNode) { ... }     and down below, always set 'tileSetNode = tileSetNode->NextSiblingElement'.  That way, when we're out of sibling elements, tileSetNode will be nullptr, and the loop will terminate.